### PR TITLE
Close popup if combobox is not visible

### PR DESCRIPTION
**Pull request by kaminova**
## What does the pull request do?

Drop-down list of the combobox is closed if the combobox becomes invisible.

## What is the current behavior?

Opened drop-down list stayed open when the combobox becomes invisible. 

## How was the solution implemented (if it's not obvious)?

I added IsVisible property change subscriptions for visual parents of the combobox.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes

None
## Obsoletions / Deprecations

None

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=6404) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=6404) before we can accept your contribution.<br/><br/>:x: kaminova [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=6404)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=6404) it.</sub>

**Comment by MarchingCube** [comment]
It seems that WPF implements this via mouse capture:
https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/ComboBox.cs#L1690-L1696

Looks like a hack based on comments and it was perhaps planned for other controls as well.

**Comment by robloo** [comment]
Jevan Saks is still at Microsoft and active on the WinUI repo. If you are curious of a better way or the thinking at the time he might answer. (The comments indicate he was assigned the internal issue).

File: src/Avalonia.Controls/ComboBox.cs [hunk 1]
```diff
@@ -291,6 +293,7 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
 
             _popup = e.NameScope.Get<Popup>("PART_Popup");
             _popup.Opened += PopupOpened;
+            _popup.Closed += PopupClosed;
```
Comment (grokys) [inline src/Avalonia.Controls/ComboBox.cs:296]: Huh, wow, looks like we weren't subscribing to the `Closed` event, meaning that `PopupClosed` was never called!

**Review by grokys — APPROVED** [review]
I think this is as good a way of any of handling this, LGTM, and you fixed a bug too ;)

**Comment by maxkatz6** [comment]
@kaminova can you resolve conflict, so this PR can be merged?

**Pull request merged**